web/src/scheduled_messages_overlay_ui.js

```diff
@@ -16,8 +16,15 @@ import * as timerender from "./timerender";
 export const keyboard_handling_context = {
     get_items_ids() {
         const scheduled_messages_ids = [];
-        for (const message of scheduled_messages.scheduled_messages_data) {
-            scheduled_messages_ids.push(message.scheduled_message_id);
+        for (const key in scheduled_messages.scheduled_messages_data) {
+            if (
+                Object.prototype.hasOwnProperty.call(
+                    scheduled_messages.scheduled_messages_data,
+                    key,
+                )
+            ) {
+                scheduled_messages_ids.push(key);
+            }
```

Comment:
I realized this would be incorrect now since we not longer have `scheduled_messages_data` sorted by default. So, `get_items_ids` should really be`get_first_item_id` and we can just use it in `scheduled_messages_overlay_ui.lauch`.

Reply:
Then we also need to sort the messages before returning the first message id in get_first_item_id()

Reply:
Yeah. So, extracting that sort function from format would be a good step too.

Reply:
I have extracted the sort function from `format` but have not changed the implementation of get_items_ids as its being used in message_overlay_ui.js also.

Reply:
Okay!